File: my-folder/0264-ugly-number-ii/solution.py
class Solution:
    def nthUglyNumber(self, n: int) -> int:
        ans=[1]
        prod_2=prod_3=prod_5=0
        for i in range(1,n):
            a=ans[prod_2]*2
            b=ans[prod_3]*3
            c=ans[prod_5]*5
            m=min(a,b,c)
            ans.append(m)
            if m==a:
                prod_2+=1
            if m==b:
                prod_3+=1
            if m==c:
                prod_5+=1
        return ans[-1]        

File: my-folder/0264-ugly-number-ii/solution.cpp
class Solution {
public:
    int nthUglyNumber(int n) {
        int a = 0, b = 0, c = 0;
        vector<int> res;
        res.push_back(1);
        for(int i = 1; i < n; i++) {
            int tmp = min(res[a]*2, min(res[b]*3, res[c]*5));
            res.push_back(tmp);
            if(tmp == res[a]*2) a++;
            if(tmp == res[b]*3) b++;
            if(tmp == res[c]*5) c++;
        }
        return res[n-1];
    }
};
